storage/sub_query.go

```diff
@@ -0,0 +1,45 @@
+package storage
+
+type SubQuery int
+
+const (
+	QueryForAllLastOperationsPerResource SubQuery = iota
+	QueryForNonResourcelessOperations
+)
+
+// The aforementioned sub-queries are dedicated to be used with ByIDExists/ByIDNotExists Criterion to allow additional querying/filtering on top of
+// combined with main query generated from the criteria.
+// As in sql's EXIST/ NOT EXIST,	in order for main query to work with the sub-query,the  sub-queries will require a where clause that compares the id from
+// the parent query with and id retrieved from the sub-query.
+//
+// Example: Get all internal operations which aren't orphans (have corresponding resources) using ByIdExist criterion:
```

Comment:
simplify the example - no need for additional criterias. in addition, make sure the example works, and add the way to use it (e.g. repo.Count/Delete/.. with the relevant object type)